Index: ocaml-4.05/stdlib/Makefile
===================================================================
--- ocaml-4.05.orig/stdlib/Makefile
+++ ocaml-4.05/stdlib/Makefile
@@ -127,8 +127,8 @@ camlheaderi target_camlheaderi
 ifeq "$(HASHBANGSCRIPTS)" "true"
 $(CAMLHEADERS): ../config/Makefile
 	for suff in '' d i; do \
-	  echo '#!$(BINDIR)/ocamlrun'$$suff > camlheader$$suff && \
-	  echo '#!$(TARGET_BINDIR)/ocamlrun'$$suff >target_camlheader$$suff; \
+	  echo '#!/usr/bin/env ocamlrun'$$suff > camlheader$$suff && \
+	  echo '#!/usr/bin/env ocamlrun'$$suff >target_camlheader$$suff; \
 	done && \
 	echo '#!' | tr -d '\012' > camlheader_ur;
 else # Hashbang scripts not supported
